### PR TITLE
First Push

### PR DESCRIPTION
**Pull request by abhishekkrsp**
# Implementations
>- auto enable and disable submit button
>- set allowed characters for textfield
>- gender is optional input
>- gender options is according to enum cases
>- used segment control for gender input

# Clicks
Screenshot 2022-03-24 at 9 52 04 AM
Screenshot 2022-03-24 at 9 52 32 AM
Screenshot 2022-03-24 at 9 52 35 AM

### CODE REVIEW

File: Form/Form/User.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//
+//  User.swift
+//  Form
+//
+//  Created by Abhishek Kumar on 23/03/22.
+//
+
+import Foundation
+
+class User {
```
Comment (priyal-p) [inline Form/Form/User.swift:10]: Do we need this to be `class`?

File: Form/Form/User.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ) {
+        self.firstname = firstName
+        self.lastName = lastName
+        self.gender = gender
+        self.age = age
+    }
+}
+
+enum Gender: String,CaseIterable {
+//    case None
```
Comment (priyal-p) [inline Form/Form/User.swift:26]: Remove the comments if not required

File: Form/Form/DisplayController.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import Foundation
+import UIKit
+
+protocol DisplayDetails: AnyObject {
+    func returnName() -> String
+    func returnAge() -> String
+    func returnGender() -> String
+}
+class UIDisplayController: UIViewController {
```
Comment (priyal-p) [inline Form/Form/DisplayController.swift:16]: Are we using it?

File: Form/Form/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        
+        guard let dataFirstName = dataFirstName, let dataLastName = dataLastName, let dataAge = dataAge else {
+            submit.isEnabled = false
+            return
+        }
+        
+        if dataFirstName.isEmpty || dataLastName.isEmpty || dataAge <= 20 {
+            submit.isEnabled = false
+            return
+        }
```
Comment (priyal-p) [inline Form/Form/ViewController.swift:73]: We can club these conditions

Comment (abhishekkrsp) [reply Form/Form/ViewController.swift:73]: Ok

File: Form/Form/DisplayController.swift (hunk 3 above)
Comment (abhishekkrsp) [reply Form/Form/DisplayController.swift:16]: No

File: Form/Form/User.swift (hunk 2 above)
Comment (abhishekkrsp) [reply Form/Form/User.swift:26]: Ok

File: Form/Form/User.swift (hunk 1 above)
Comment (abhishekkrsp) [reply Form/Form/User.swift:10]: Not now

**Pull request merged**